src/clients/GithubClient.ts

```diff
@@ -0,0 +1,282 @@
+  id: number;
+  tagName: string;
+  publishedAt: string | null;
+}
+
+export class GitHubClient implements GitHubApiClient {
+  private baseUrl: string;
+  private headers: Record<string, string>;
+
+  constructor(options: GitHubClientOptions = {}) {
+    this.baseUrl = options.baseUrl || "https://api.github.com";
+    this.headers = {
+      Accept: "application/vnd.github+json",
+      "User-Agent": "astro-github-client",
+    };
+
+    if (options.token) {
+      this.headers["Authorization"] = `token ${options.token}`;
+    }
+  }
+
+  async request<T>(
+    endpoint: string,
+    params: Record<string, string | number | boolean> = {},
+  ): Promise<T> {
+    const url = new URL(`${this.baseUrl}${endpoint}`);
+
+    Object.entries(params).forEach(([key, value]) => {
+      if (value !== undefined && value !== null) {
+        url.searchParams.append(key, String(value));
+      }
+    });
+
+    try {
+      const response = await fetch(url.toString(), { headers: this.headers });
+
+      if (!response.ok) {
+        const errorData: GitHubErrorResponse = await response
+          .json()
+          .catch(() => ({
+            message: `HTTP error ${response.status}`,
+          }));
+        throw new Error(
+          errorData.message || `GitHub API error: ${response.status}`,
+        );
+      }
+
+      return (await response.json()) as T;
+    } catch (error) {
+      console.error("GitHub API request failed:", error);
+      throw error;
+    }
+  }
+
+  async getLatestRelease(user: string, repo: string): Promise<LatestRelease> {
+    const response = await this.request<GitHubRelease>(
+      `/repos/${user}/${repo}/releases/latest`,
+    );
+    return {
+      id: response.id,
+      htmlUrl: response.html_url,
+      tagName: response.tag_name,
+      publishedAt: response.published_at,
+      description: response.body?.replace(/\n/g, "<br>") ?? null,
+      sourceTar: response.tarball_url,
+      sourceZip: response.zipball_url,
+      assets: response.assets.map((a) => ({
+        name: a.name,
+        downloadCount: a.download_count,
+        size: a.size,
+        browserDownloadUrl: a.browser_download_url,
+      })),
+    };
+  }
+
+  async getReleases(user: string, repo: string): Promise<Release[]> {
+    const response = await this.request<GitHubRelease[]>(
+      `/repos/${user}/${repo}/releases`,
+    );
+    return response.map((r) => ({
+      id: r.id,
+      tagName: r.tag_name,
+      publishedAt: r.published_at,
+    }));
+  }
+
+  async getReadme(user: string, repo: string): Promise<String> {
+    const response = await this.request<GitHubReadme>(
+      `/repos/${user}/${repo}/readme`,
+    );
+    const content = new TextDecoder().decode(
+      Uint8Array.from(atob(response.content), (c) => c.charCodeAt(0)),
+    );
+    const formatted = content.replace(/\n/g, "<br>");
+    console.log(formatted);
+    return formatted;
+  }
+}
+
+const defaultClient = new GitHubClient();
+export default defaultClient;
```

Comment:
A i jeszcze nie używajmy default exportów, bo nie mają żadnych plusów nad named exportami

Więc żeby wyexportować tylko singleton
```suggestion
export { defaultClient as GithubClient };
```

Oraz usunąć export wyżej z klasy